bench/Chainweb/Pact/Backend/Bench.hs

```diff
@@ -57,8 +60,33 @@ import Chainweb.Utils.Bench
 import Chainweb.Utils (sshow)
 import Chainweb.Version
 
-v :: ChainwebVersion
-v = fastForkingCpmTestVersion petersonChainGraph
+testVer :: ChainwebVersion
+testVer = fastForkingCpmTestVersion petersonChainGraph
+
+testChainId :: ChainId
+testChainId = unsafeChainId 0
+
+genesisPc :: ParentContext
+genesisPc = genesisParentContext testVer testChainId
+
+-- allowing a straightforward list of blocks to be passed to the API,
+-- and only exposing the PactDbEnv part of the block context.
+-- also present in Chainweb.Test.Pact.Checkpointer
+cpRestoreAndSave
+  :: (Monoid q)
+  => Checkpointer logger
+  -> ParentContext
+  -> [(ParentContext, ChainwebPactDbEnv logger -> IO q)]
+  -> IO q
+cpRestoreAndSave cp pc blks = snd <$> _cpRestoreAndSave cp pc
+  (traverse Stream.yield [0..length blks - 1])
+  (\dbenv _pc i -> fmap (,fst (blks !! i)) $ snd (blks !! i) (_cpPactDbEnv dbenv))
+
+childOf :: ParentContext -> BlockHash -> ParentContext
```

Comment:
Just an observation, not necessarily a recommendation to change something,

I notice when I read `ParentContext` I keep thinking "the context of the parent", when really the meaning is "the parent value available in the current block's context").

By contrast, `ContextParent` makes me immediately think "the parent value available in the current block's context".

Reply:
This is fair, this and the earlier comment make me consider renaming `ParentContext` to `ParentHeaderInfo` or something.

Reply:
Deleted this type entirely in the end.